pkg/clients/kubernetes.go

```diff
@@ -49,6 +50,27 @@ func KubernetesConfig(log *logging.Logger) (*rest.Config, error) {
 	return instances.KubernetesConfig, nil
 }
 
+// GetSecretData - Returns the data inside of a given secret
+func GetSecretData(secretName, namespace string) (map[string][]byte, error) {
+	var log logging.Logger
+	k8scli, err := Kubernetes(&log)
+	if err != nil {
+		return nil, err
+	}
+	var ret = make(map[string][]byte)
+
+	secretData, err := k8scli.CoreV1().Secrets(namespace).Get(secretName, meta_v1.GetOptions{})
+	if err != nil {
+		log.Error("Unable to load secret '%s' from namespace '%s'", secretName, namespace)
+		return ret, nil
+	}
+	log.Debug("Found secret with name %v\n", secretName)
+
+	ret = secretData.Data
+
+	return ret, nil
```

Comment:
feels weird to assign a value to ret just to return it on the next line, I'd expected: `return secretData.Data` I'd get rid of ret entirely. Because you're creating an array that you ultimately throwaway on every successful call. I'd rather create an array only during the error condition.

```golang
// GetSecretData - Returns the data inside of a given secret
func GetSecretData(secretName, namespace string) (map[string][]byte, error) {
    var log logging.Logger
    k8scli, err := Kubernetes(&log)
    if err != nil {
        return nil, err
    }

    secretData, err := k8scli.CoreV1().Secrets(namespace).Get(secretName, meta_v1.GetOptions{})
    if err != nil {
        log.Errorf("Unable to load secret '%s' from namespace '%s'", secretName, namespace)
        return make(map[string][]byte), nil
    }
    log.Debugf("Found secret with name %v\n", secretName)

    return secretData.Data, nil
```